js/review.js

```diff
@@ -91,11 +91,31 @@ $(function () {
         if (checkForm()) {
             var tweetUri = new TweeFiUtils.TweetUri($('#tweet_url').val());
             var data = createReview(tweetUri);
-            submitReview(data, tweetUri).then(function (reviewURI) {
+            submitReview(data, tweetUri).then(function (submitResult) {
                 $("#cr-valid").removeClass("hide").addClass("show");
-                console.log("Forwarding to:  " + reviewURI);
-                window.location.href = reviewURI;
+                let firstReview = submitResult.firstReview;
+                if (firstReview) {
+                    $("#reviewURI").html(submitResult.rootValue + "publictwee-fi/" + tweetUri.getUser() + "/" + tweetUri.getStatus());
+                    $("#repoURI").html(submitResult.rootValue);
+                    $("#reviewURI").attr("href", submitResult.rootValue + "publictwee-fi/" + tweetUri.getUser() + "/" + tweetUri.getStatus());
+                    $("#repoURI").attr("href", submitResult.rootValue);
+                    $("#firsttime").removeClass("invisible").addClass("visible");
+                }
+                if (!firstReview) {
+                    $("#cr-valid").html("ClaimReview successfully created. Redirecting...");
+                    console.log("Forwarding to: " + submitResult.reviewUri);
+                    setTimeout(window.location.href = submitResult.reviewUri, "5000");
+                }
+                $("#dismissFirsttime").on("click", () => {
+                    $("#firsttime").removeClass("visible").addClass("visible");
+                    $("#cr-valid").removeClass("alert-success").addClass("alert-danger");
+                    $("#cr-valid").html("Redirecting...");
+                    console.log("Forwarding to: " + submitResult.reviewUri);
+                    setTimeout(window.location.href = submitResult.reviewUri, "10000");
+                });
+
```

Comment:
The above could have the following   with promises:
```
(firstReview ? showFirstReviewInfo() : showSucccessConfirmation()).then(() => window.location.href = submitResult.reviewUri);
```
I think this would be more elegant and readable

Reply:
It just occurs to me, that you're using setTimeout wrongly, as you're not apssing a function as argument